src/dashboard/Data/Browser/Browser.react.js

```diff
@@ -68,68 +99,65 @@ export default class Browser extends DashboardView {
     if (!this.props.params.className && this.props.schema.data.get('classes')) {
       this.redirectToFirstClass(this.props.schema.data.get('classes'));
     } else if (this.props.params.className) {
-      if (this.props.location.query && this.props.location.query.filters) {
-        let filters = new List();
-        let queryFilters = JSON.parse(this.props.location.query.filters);
-        queryFilters.forEach((filter) => {
-          filters = filters.push(new Map(filter));
-        });
-        this.setState({ filters }, () => {
-          this.fetchData(this.props.params.className, this.state.filters);
-        });
-      } else {
-        this.fetchData(this.props.params.className, this.state.filters);
-      }
-      this.setState({
-        ordering: ColumnPreferences.getColumnSort(
-          false,
-          this.context.currentApp.applicationId,
-          this.props.params.className
-        )
-      });
+      this.prefetchData(this.props, this.context);
     }
   }
 
   componentWillReceiveProps(nextProps, nextContext) {
     if (this.context !== nextContext) {
-      let changes = {
-        filters: new List(),
-        data: null,
-        newObject: null,
-        lastMax: -1,
-        ordering: ColumnPreferences.getColumnSort(
-            false,
-            nextContext.currentApp.applicationId,
-            nextProps.params.className
-        ),
-        selection: {},
-        relation: null
-      };
-      //TODO: url limit issues ( we may want to check for url limit), unlikely but possible to run into
-      if (nextProps.location.query && nextProps.location.query.filters) {
-        let queryFilters = JSON.parse(nextProps.location.query.filters);
-        queryFilters.forEach((filter) => {
-          changes.filters = changes.filters.push(new Map(filter));
-        });
-      }
-
       if (this.props.params.appId !== nextProps.params.appId || !this.props.params.className) {
-        changes.counts = {};
+        this.setState({ counts: {} });
         Parse.Object._clearAllState();
       }
-      this.setState(changes);
-      if (nextProps.params.className) {
-        this.fetchData(nextProps.params.className, nextProps.location.query && nextProps.location.query.filters ? changes.filters : []);
-      }
+      this.prefetchData(nextProps, nextContext);
       nextProps.schema.dispatch(ActionTypes.FETCH)
       .then(() => this.handleFetchedSchema());
-
     }
     if (!nextProps.params.className && nextProps.schema.data.get('classes')) {
       this.redirectToFirstClass(nextProps.schema.data.get('classes'));
     }
   }
 
+  async prefetchData(props, context) {
```

Comment:
We aren't using `async`/`await` yet because it still isn't in the standard. Can you stick to Promises for now?